core/src/test/java/org/apache/iceberg/TestScanSummary.java

```diff
@@ -32,7 +33,47 @@
 import static org.apache.iceberg.expressions.Expressions.lessThan;
 import static org.apache.iceberg.expressions.Expressions.lessThanOrEqual;
 
-public class TestScanSummary {
+public class TestScanSummary extends TableTestBase {
+
+  @Test
+  public void testSnapshotTimeRangeValidation() {
+    long t0 = System.currentTimeMillis();
+
+    table.newAppend()
+        .appendFile(FILE_A) // data_bucket=0
+        .appendFile(FILE_B) // data_bucket=1
+        .commit();
+
+    long t1 = System.currentTimeMillis();
+
+    table.newAppend()
+        .appendFile(FILE_C) // data_bucket=2
+        .commit();
+
+    long secondSnapshotId = table.currentSnapshot().snapshotId();
+
+    long t2 = System.currentTimeMillis();
+
+    // expire the first snapshot
+    table.expireSnapshots()
+        .expireOlderThan(t1)
+        .commit();
+
+    Assert.assertEquals("Should have one snapshot",
+        1, Lists.newArrayList(table.snapshots()).size());
```

Comment:
From our internal builds, this assert fails intermittently when `t0 == t1`.

Reply:
@xabriel @rdblue we also see failures from time to time in our internal builds

Reply:
```
java.lang.AssertionError: No exception was thrown (Should fail summary because range may include expired snapshots), expected: java.lang.IllegalArgumentException
```

Reply:
Okay, I'll update it to make sure the timestamps aren't equal.

Reply:
I opened a PR for this: https://github.com/apache/incubator-iceberg/pull/433